client/ayon_core/hosts/houdini/plugins/publish/collect_inputs.py

```diff
@@ -108,13 +107,32 @@ def process(self, instance):
             )
             return
 
-        # Collect all upstream parents
-        nodes = list(iter_upstream(output))
-        nodes.append(output)
-
-        # Collect containers for the given set of nodes
-        containers = collect_input_containers(nodes)
+        # For large scenes the querying of "host.ls()" can be relatively slow
+        # e.g. up to a second. Many instances calling it easily slows this
+        # down. As such, we cache it so we trigger it only once.
+        # todo: Instead of hidden cache make "CollectContainers" plug-in
+        cache_key = "__cache_containers"
+        scene_containers = instance.context.data.get(cache_key, None)
```

Comment:
just a small question, is the `cache_key` always named as `__cache_containers`? It seems quite hard-coded to me but if it is always named like this, it is still okay.

Reply:
Good question. In the past I had sometimes prefixed it also with the class name to ensure better uniqueness. But yes, in a way it is hardcoded. Feel too unsafe? Should I prefix it more?

Reply:
> @moonyuet let me know what you'd like me to do here to make this ready for merging

we can merge this first as I guess there would be enhancement when building the universal node for publishing.